port/osx/omrosdump.c

```diff
@@ -210,10 +210,10 @@ coredump_to_file(mach_port_t task_port, pid_t pid)
 			}
 			if (to_read != data_size) {
 				break;
```

Comment:
If a segment is not read in its entirety (e.g. via the `break` here), we need
```
    segments[i].filesize = bytes_read;
```
after line 223 (as was present as a comment in ef5cdbaa01c7628a4d96ade836f1565d37906d80).

Reply:
I originally thought I needed the line `segments[i].filesize = bytes_read;` for the case `bytes_read < segments[i].filesize`.
But I removed the line because it caused UnsatisfiedLinkErrors in an x86_64 test job as I described above.

Reply:
Without it, there's the possibility that `segments[i].filesize` is too large, and the segment will extend into the file space for following segments (or beyond the end of the file).

Reply:
Perhaps the `UnsatisfiedLinkError` will be resolved by adoptium/aqa-tests#3559?

Reply:
Only if running JCKs. We'd have to add the option to the functionality test(s). Plus that is a jdk8 problem afaik.

We ran a number of tests for this specific issue, see runtimes/backlog/issues/742#issuecomment-42417615, but they didn't run on mac11x64rt01. It's possible there is a machine problem. We have had missing prereqs in the past on some platforms runtimes/backlog/issues/742#issuecomment-42407773. There is a list of the prereqs required for jdk18 Mac at runtimes/infrastructure/issues/5416#issuecomment-42508170.

Reply:
I added `segments[i].filesize = bytes_read;` back again.

Reply:
I ran extended.functional on mac10x64rt03 and on osxrt2, and got no `UnsatisfiedLinkError`.